src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerOptions.cs

```diff
@@ -14,21 +14,12 @@ public class KestrelServerOptions
 
         /// <summary>
         /// Gets or sets value that instructs <seealso cref="KestrelServer"/> whether it is safe to 
-        /// pool the Request and Response <seealso cref="System.IO.Stream"/> objects
+        /// cache the headers and streams
         /// for another request after the Response's OnCompleted callback has fired. 
-        /// When this values is greater than zero, it is not safe to retain references to feature components after this event has fired.
-        /// Value is zero by default.
+        /// When this values is true, it is not safe to retain references to feature components after this event has fired.
+        /// Value is false by default.
         /// </summary>
-        public int MaxPooledStreams { get; set; }
-
-        /// <summary>
-        /// Gets or sets value that instructs <seealso cref="KestrelServer"/> whether it is safe to 
-        /// pool the Request and Response headers
-        /// for another request after the Response's OnCompleted callback has fired. 
-        /// When this values is greater than zero, it is not safe to retain references to feature components after this event has fired.
-        /// Value is zero by default.
-        /// </summary>
-        public int MaxPooledHeaders { get; set; }
+        public bool CacheHeadersAndStreams { get; set; }
```

Comment:
I don't understand why that wouldn't be addressed by the same issue. The idea is to get rid of a first class notion of caching 2 things that handle "the lifetime of things kestrel allocates via some extensibility point". This is the first 2 of many things we'll want to enable pooling for. Are we going to keep adding flags:

```C#
public bool CacheHeadersAndStreams { get; set; }
public bool CacheWriteContexts { get; set; }
public bool CacheSomeotherStuff { get; set; }

```